great_expectations/rule_based_profiler/helpers/util.py

```diff
@@ -1080,3 +1091,17 @@ def sanitize_parameter_name(
         name = f"{name}{FULLY_QUALIFIED_PARAMETER_NAME_SEPARATOR_CHARACTER}{suffix}"
 
     return name.replace(FULLY_QUALIFIED_PARAMETER_NAME_SEPARATOR_CHARACTER, "_")
+
+
+class _NumericIterableWithDtype(Iterable, Protocol):
+    @property
+    def dtype(self) -> Any:
+        ...
+
+
+def _is_iterable_numeric_and_has_dtype_attr(
```

Comment:
```suggestion
def _is_iterable_of_numeric_dtypes(
```
Just thinking about this a bit more. We are doing more than just checking if it has the `dtype` attr aren't we.
If we were just checking for the attr the type-guard would just be...
```python
if hasattr(obj, "dtype") and isinstance(obj, Iterable):
   return True
return False
```